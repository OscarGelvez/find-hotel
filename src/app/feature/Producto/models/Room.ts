export interface IRoom {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Objeto hotel al cual pertenece la habitación
   */
  hotel: IHotel;
  /**
   * Valor de la habitación
   */
  value: number;
  /**
   * Indica si tiene o no aire acondicionado
   */
  ac: string;
  /**
   * Indica si cuenta o no con parquedadero
   */
  parking: string;
  /**
   * Fecha inicial de disponibilidad
   */
  available_from: string;
  /**
   * Fecha final de disponibilidad
   */
  available_until: string;
  /**
   * Indica si cuenta con conexión wifi
   */
  wifi: string;
  /**
   * Capacidad de personas por habitación
   */
  capacity: number;
  /**
   * Descripción general de la habitación
   */
  description: string;
}

export interface IHotel {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Nombre del hotel
   */
  name: string;
  /**
   * Dirección y/o ubicacion del hotel
   */
  address: string;
  /**
   * Cantidad de estrellas del hotel
   */
  stars: number;
  /**
   * Descripción general
   */
  description: string;
  /**
   * Puntaje de calificación dado por usuarios
   */
  score: number;
}
