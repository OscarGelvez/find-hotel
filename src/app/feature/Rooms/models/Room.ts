export interface IRoom {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Título de la habitación
   */
  title: string;
  /**
   * Imagen descriptiva de la habitación
   */
  image: string;
  /**
   * Objeto hotel al cual pertenece la habitación
   */
  hotel: IHotel;
  /**
   * Valor de la habitación
   */
  value: string;
  /**
   * Indica si tiene o no aire acondicionado
   */
  ac: string;
  /**
   * Indica si cuenta o no con parquedadero
   */
  parking: string;
  /**
   * Fecha inicial de disponibilidad
   */
  available_from: string;
  /**
   * Fecha final de disponibilidad
   */
  available_until: string;
  /**
   * Indica si cuenta con conexión wifi
   */
  wifi: string;
  /**
   * Capacidad de personas por habitación
   */
  capacity: number;
  /**
   * Descripción general de la habitación
   */
  description: string;
}

export interface IHotel {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Nombre del hotel
   */
  name: string;
  /**
   * Dirección y/o ubicacion del hotel
   */
  address: string;
  /**
   * Cantidad de estrellas del hotel
   */
  stars: number;
  /**
   * Descripción general
   */
  description: string;
  /**
   * Puntaje de calificación dado por usuarios
   */
  score: number;
}

export interface IFieldsForm {
  /**
   * Fecha inicio de reservación
   */
  fromDate?: string;
  /**
   * Fecha final de reservación
   */
  untilDate?: string;
  /**
   * Cantidad de personas en habitación
   */
  capacity?: string;
  /**
   * Habitación con aire acondicionado
   */
  withAir?: boolean;
  /**
   * Habitación con parqueadero propio
   */
  withParking?: boolean;
}

export interface IFieldsFormBooking {
  /**
   * Nombre de ususario que hace la reserva
   */
  name: string;
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}

export interface IBookingData {
  /**
   * Objeto con información de la habitación seleccionada
   */
  roomData: IRoom;
  /**
   * Objeto con la información el usuari oque reserva
   */
  bookData: IFieldsFormBooking;
}

export interface IFieldsFormFindBooking {
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}
