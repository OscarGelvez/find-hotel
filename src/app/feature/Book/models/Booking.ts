import { IRoom } from 'app/feature/Rooms/models/Room';

export interface IFieldsFormBooking {
  /**
   * Nombre de ususario que hace la reserva
   */
  name: string;
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
  /**
   * Fecha de ingreso al hotel
   */
  fromDateBook: string;
  /**
   * Fecha de salida del hotel
   */
  untilDateBook: string;
}

export interface IBooking {
  /**
   * Objeto con información de la habitación seleccionada
   */
  roomData: IRoom;
  /**
   * Objeto con la información el usuari oque reserva
   */
  bookData: IFieldsFormBooking;
  /**
   * Identificador en base de datos
   */
  id?: number;
}

export interface IFieldsFormFindBooking {
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}
