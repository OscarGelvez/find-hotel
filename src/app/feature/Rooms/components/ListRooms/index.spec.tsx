import '@testing-library/jest-dom';

import { cleanup,render } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import { IRoom } from '../../models/Room';
import { ListRooms } from './index';

const newRoom: IRoom = {
  id: 50,
  title: 'Suite full. Caroline',
  image: 'https://static.abc.es/Media/201504/27/hotel12--644x362.jpg',
  hotel: {
    id: 1,
    name: 'Gold Hotel 5',
    address: 'Calle 54 # 56 -32 Av Libre',
    stars: 4,
    description:
      'Hotel con ambiente familiar, increible opción. Muchas comodidades',
    score: 8.7,
  },
  value: '154.000',
  ac: 'Y',
  parking: 'Y',
  available_from: '2021-04-13',
  available_until: '2021-04-30',
  wifi: 'Y',
  capacity: 1,
  description: 'Baño privado, cama doble, con nevera, Wifi, TV.',
};
afterEach(cleanup);

it('rederizar el componente ListRooms sin errores', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <ListRooms listRooms={[]} loadDetailRoom={() => null} />,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renderiza el componente ListRooms y verifica existencia de componentes hijos', () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <ListRooms listRooms={[newRoom]} loadDetailRoom={() => null} />
    </BrowserRouter>
  );

  expect(getByTestId('list-rooms')).toContainElement(
    getByTestId('card-information')
  );
});

it('Compara snapshot del componente ListRooms cuando hay habitaciones', () => {
  const element = renderer
    .create(
      <BrowserRouter>
        <ListRooms listRooms={[newRoom]} loadDetailRoom={() => null} />
      </BrowserRouter>
    )
    .toJSON();
  expect(element).toMatchSnapshot();
});

it('Compara snapshot del componente ListRooms cuando no hay habitaciones', () => {
  const element = renderer
    .create(<ListRooms listRooms={[]} loadDetailRoom={() => null} />)
    .toJSON();
  expect(element).toMatchSnapshot();
});
