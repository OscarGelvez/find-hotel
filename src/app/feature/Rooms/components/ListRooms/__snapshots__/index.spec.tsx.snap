// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compara snapshot del componente ListRooms cuando hay habitaciones 1`] = `
<div
  className="card-rooms"
  data-testid="list-rooms"
>
  <div
    className="row pt-5"
  >
    <div
      className="col"
    >
      <div
        className="card room-item"
        data-testid="card-information"
      >
        <div
          className="card-header room-item-header "
        >
          <h3>
            Gold Hotel 5
          </h3>
          <h4>
            8.7
          </h4>
        </div>
        <div
          className="card-body"
        >
          <div
            className="row"
          >
            <div
              className="col col-md-4"
            >
              <img
                alt="hotel room"
                className="img-fluid"
                height="180px"
                src="https://static.abc.es/Media/201504/27/hotel12--644x362.jpg"
                width="260px"
              />
            </div>
            <div
              className="col col-md-8"
            >
              <div
                className="row"
              >
                <div
                  className="col-12 col-lg-8"
                >
                  <h5
                    className="card-title"
                  >
                    Suite full. Caroline
                  </h5>
                </div>
                <div
                  className="col-12 col-lg-4"
                >
                  <a
                    className="sc-AxjAm jqjyPd"
                    href="/detail"
                    onClick={[Function]}
                  >
                    <button
                      className="btn btn-primary btn-book w-100"
                      data-testid="form-button-load-detail"
                      onClick={[Function]}
                      type="button"
                    >
                      <i
                        className="far fa-calendar-check ml-3"
                      />
                       Seleccionar
                    </button>
                  </a>
                </div>
                <div
                  className="col-12 col-md-12"
                >
                  <h4>
                    <i
                      className="far fa-money-bill-alt"
                    />
                     
                    154.000
                  </h4>
                </div>
                <div
                  className="col-12 col-md-12 d-flex"
                >
                  <h6
                    className="font-weight-bold"
                    data-testid="custom-field"
                  >
                    Inicio
                     
                    <i
                      className="fas fa-calendar-alt"
                    />
                     :
                     
                    2021-04-13
                  </h6>
                  <h6
                    className="font-weight-bold"
                    data-testid="custom-field"
                  >
                    -  Fin
                     
                    <i
                      className="fas fa-calendar-alt"
                    />
                     :
                     
                    2021-04-30
                  </h6>
                </div>
              </div>
              <p
                className="card-text"
              >
                Baño privado, cama doble, con nevera, Wifi, TV.
              </p>
              <div
                className="row"
              >
                <div
                  className="col col-md-6"
                >
                  <h6
                    className="font-weight-bold"
                    data-testid="custom-field"
                  >
                    camas
                     
                    <i
                      className="fas fa-bed"
                    />
                     :
                     
                    1
                  </h6>
                  <h6
                    className="font-weight-bold"
                    data-testid="custom-field"
                  >
                    Aire
                     
                    <i
                      className="fas fa-wind"
                    />
                     :
                     
                    <i
                      className="fas fa-check"
                    />
                  </h6>
                  <h6
                    className="font-weight-bold"
                    data-testid="custom-field"
                  >
                    Wifi
                     
                    <i
                      className="fas fa-wifi"
                    />
                     :
                     
                    <i
                      className="fas fa-check"
                    />
                  </h6>
                </div>
                <div
                  className="col col-md-6"
                >
                  <h6
                    className="font-weight-bold"
                    data-testid="custom-field"
                  >
                    Parqueadero
                     
                    <i
                      className="fas fa-car"
                    />
                     :
                     
                    <i
                      className="fas fa-check"
                    />
                  </h6>
                  <h6
                    className="font-weight-bold"
                    data-testid="custom-field"
                  >
                    Baños
                     
                    <i
                      className="fas fa-restroom"
                    />
                     :
                     
                    <i
                      className="fas fa-check"
                    />
                  </h6>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Compara snapshot del componente ListRooms cuando no hay habitaciones 1`] = `
<div
  className="card-rooms"
  data-testid="list-rooms"
>
  <div
    className="row pt-5"
  >
    <div
      className="col"
    />
  </div>
</div>
`;
